CMakeLists.txt

```diff
@@ -1,5 +1,5 @@
 # Check CMake version
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.20)
```

Comment:
Not being very knowledgeable regarding CMake, is there a reason for increasing the minimum required version in support of pFUnit testing?

Reply:
The `pFUnit` library uses a naming convention for its submodule dependencies that is not fully supported by versions of `CMake` older than `3.20`. Specifically, it names its dependency root directories using the `<dependency_name>_ROOT` format, which requires `CMake policy CMP0074` to be set when using `CMake` versions earlier than `3.20`.

Below is an example of the warning message produced when compiling obs2ioda with `CMake 3.10`:

```
CMake Warning (dev) at /usr/share/cmake/Modules/CMakeFindDependencyMacro.cmake:76 (find_package):
  Policy CMP0074 is not set: find_package uses <PackageName>_ROOT variables.
  Run "cmake --help-policy CMP0074" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  CMake variable GFTL_ROOT is set to:

    /home/astokely/projects/spack-stacks/spack-stack_1.8.0/envs/mpas-bundle.alta.gcc/install/gcc/11.4.1/gftl-1.14.0-h4pxvm3/GFTL-1.14/cmake

  For compatibility, CMake is ignoring the variable.
  ```

Reply:
GitHub comments probably aren't the best place for me to try to get a CMake education, but the output of `cmake --help-policy CMP0074` suggests that the version threshold is 3.12, and it looks like [pFUnit requires CMake 3.12](https://github.com/Goddard-Fortran-Ecosystem/pFUnit/blob/main/CMakeLists.txt#L25). So should the requirement be 3.12 rather than 3.20? Apologies if I'm off base here!

Reply:
I think this is a good idea. I'll change the minimum version to 3.12.